JavaOOPHomework1/src/PurchaseManager.java

```diff
@@ -0,0 +1,33 @@
+import java.time.LocalDate;
+
+/**
+ * The class should hold the processPurchase(Product product, Customer customer) static method that handles purchases
+ * (takes money from customer, reduces product quantity by 1).
+ * The PurchaseManager should print descriptive messages in the following situations:
+ *         •    If the product is out of stock (i.e. no quantity)
+ *         •    If the product has expired  (optional)
+ *         •    If the buyer does not have enough money
+ *         •    If the buyer does not have permission to purchase the given product
+ */
+public class PurchaseManager {
+
+    public static LocalDate processPurchase(Product product, Customer customer) {
+
+        if (customer.getBalance() - product.getPrice() < 0) {
```

Comment:
atunci cand una dintre reguli este incalcata pe langa afisarea mesajului trebuie sa opresti firul de executie (pentru moment folosind return). In implementarea curenta dupa ce se afiseaza unul dintre mesaje se trece mai departe si se face tranzactia.